### ISSUE TITLE
Generate a unique name from user input to use with multiple Bicep modules

### ISSUE THREAD

**Issue by brooke-hamilton** [Azure/missionlz#503]
## Benefit/Result/Outcome

We need a simple proof-of-concept for generating a unique name based on user input, i.e., the `resourcePrefix` parameter, that can be passed to individual modules in order to ensure uniqueness, and also be consistent to preserve idempotency. Idempotency and uniqueness are conflicting goals, and we believe that using a user input like `resourcePrefix` is the best compromise between the two goals.

This issue will unblock #498. 

## Acceptance Criteria
-  User input generates a unique name using the Bicep `uniqueString` function.
-  The resulting unique string is passed to a module that creates a storage account using the unique string as part of the storage account name.
- The storage account can be deployed multiple times without destroying and recreating, preserving idempotency.

**Comment by glennmusa** [comment]
@ExchMaster pointed us to the "shared variables" pattern that could prove helpful for this work:

https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/patterns-shared-variable-file#example-1-naming-prefixes

**Comment by glennmusa** [comment]
This could be OBE by the proposed changes stemming from #556.

In those proposed changes the resource prefix is prepended to resource names and the unique string is calculated for each deployment for the resources that need it (storage accounts, L28 in this example): 

https://github.com/Azure/missionlz/blob/1429d94f1824e4d3581b955479c1f5d221494578/src/bicep/mlz.bicep#L17-L31

**Comment by brooke-hamilton** [comment]
This backlog item is superseded by #556.
